packages/axelar-local-dev-cosmos/src/__tests__/contracts/Factory.sol

```diff
@@ -142,6 +188,27 @@ contract Factory is AxelarExecutable {
         return newWallet;
     }
 
+    function _createAndDeposit(
+        string memory ownerStr,
+        address tokenOwner,
+        IPermit2.PermitTransferFrom memory permit,
+        bytes memory signature
+    ) internal returns (address newWallet) {
+        require(tokenOwner != address(0), "tokenOwner=0");
+        require(permit.permitted.token != address(0), "token=0");
+        require(permit.permitted.amount > 0, "amount=0");
+
+        newWallet = _createSmartWallet(ownerStr);
+
+        IPermit2.SignatureTransferDetails memory details = IPermit2
+            .SignatureTransferDetails({
+                to: newWallet,
+                requestedAmount: permit.permitted.amount
+            });
+
+        permit2.permitTransferFrom(permit, details, tokenOwner, signature);
```

Comment:
Note: (like other clients) we rely on the permit2 contract to check the signature (and nonce and deadline)